CIP-Registration/CIP-Registration.md

```diff
@@ -0,0 +1,101 @@
+Authors: Sebastien Guillemot <sebastien@emurgo.io>, Rinor Hoxha <rinor.hoxha@iohk.io>, Mikhail Zabaluev <mikhail.zabaluev@iohk.io>
+Comments-URI: https://forum.cardano.org/t/cip-catalyst-registration-metadata-format/44038
+Status: Draft
+Type: Standards
+Created: 2020-01-05
+License: CC-BY-4.0
+---
+
+## Abstract
+
+Cardano uses a sidechain for its treasury system. One needs to "register" to participate on this sidechain by submitting a registration transaction on the mainnet chain. This CIP details the registration transaction format.
+
+## Motivation
+
+Cardano uses a sidechain for its treasury system ("Catalyst"). One of the desirable properties of this sidechain is that even if its safety is compromised, it doesn't cause loss of funds on the main Cardano chain. To achieve this, instead of using your wallet's recovery phrase on the sidechain, we need to use a brand new "voting key".
+
+However, since 1 ADA = 1 vote, a user needs to associate their mainnet ADA to their new voting key. This can be achieved through a registration transaction.
+
+We therefore need a registration transaction that serves two purposes:
+
+1. Registers a "voting key" to be included in the sidechain
+2. Associates mainnet ADA to this voting key
+
+## Specification
+
+### Voting key format
+
+A voting key is simply an ED25519 key. How this key is created is up to the wallet.
+
+### Associating stake with a voting key
+
+Recall: Cardano uses the UTXO model so to completely associate a wallet's balance with a voting key (i.e. including enterprise addresses), we would need to associate every payment key to a voting key individually. Although there are attempts at this (see [CIP8](../CIP-0008/CIP-0008.md)), the resulting data structure is a little excessive for on-chain metadata (which we want to keep small)
+
+Given the above, we choose to only associate staking keys with voting keys. Since most Cardano wallets only use base addresses for Shelley wallet types, in most cases this should perfectly match the user's wallet.
+
+### Registration metadata format
+
+A Catalyst registration transaction are a regular Cardano transaction with a specific transaction metadata associated with it
+
+Notably, there should be two entries inside the metadata map:
+
+Voting key registration
+```
+61284: {
+  // voting_key - CBOR byte array
+  1: "0xa6a3c0447aeb9cc54cf6422ba32b294e5e1c3ef6d782f2acff4a70694c4d1663",
+  // stake_pub - CBOR byte array
+  2: "0xad4b948699193634a39dd56f779a2951a24779ad52aa7916f6912b8ec4702cee"
+}
+```
+
+Signing the voting public key with the staking private key
+```
+61285: {
+  // signature - ED25119 signature CBOR byte array
+  1: "0x8b508822ac89bacb1f9c3a3ef0dc62fd72a0bd3849e2381b17272b68a8f52ea8240dcc855f2264db29a8512bfcd522ab69b982cb011e5f43d0154e72f505f007"
+}
+```
+
+This corresponds to the following CDDL definition
+
+```
+registration_cbor = {
+  61284: key_registration
+, 61285: registration_signature
```

Comment:
We're adding `61286` which is the for payment. This isn't defined yet, but will be in the next week. This is not required for others to implement, it's only used when sending funds that are tagged from catalyst. We'll have some sort of value that signifies what kind of catalyst payment it is (advisor, voter or proposal winner). We'll start tagging payments in fund3 with this data.